app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -454,8 +452,8 @@ class FireDatabase : Database() {
             val friendsListData = data.child(friendsListFile)
             val runsHistoryData = data.child(runsHistoryFile)
             val dailyGoalData = data.child(dailyGoalsFile)
-            val totalDistance = data.child(achievementsFile).child(totalDistanceFile).value
-            val totalActivityTime = data.child(achievementsFile).child(totalActivityTimeFile).value
+            val totalDistance = data.child(achievementsFile).child(totalDistanceFile).getValue(Double::class.java)
```

Comment:
The value that come from the database are of type long, are you sure that this work ?
Pastly I have an issue with that.

Reply:
When I pushed dailygoals to the database, the ones with 0 values where of type long but the ones with 3.1 for example where of type double so the casting didn't work and threw an error.
I used the solution there: https://stackoverflow.com/questions/47140436/how-to-save-zero-as-a-double-in-firebase-database

Reply:
Ok it is good then. Thank you for your clarification.